File: 1321-get-equal-substrings-within-budget/1321-get-equal-substrings-within-budget.java
class Solution {
    public int equalSubstring(String s, String t, int maxCost) {
        int left = 0, length = 0, maxLen = Integer.MIN_VALUE, total = 0;
        for(int right = 0; right < s.length(); right++){
            total += Math.abs((int)s.charAt(right) - (int)t.charAt(right));
            length++;
            while(total > maxCost){
                total -= Math.abs((int)s.charAt(left) - (int)t.charAt(left));
                length--;
                left++;
            }
            maxLen = Math.max(maxLen, length);
        }
        return maxLen;
    }
}

File: 1321-get-equal-substrings-within-budget/1321-get-equal-substrings-within-budget.rs
impl Solution {
    pub fn equal_substring(s: String, t: String, max_cost: i32) -> i32 {
        let s_bytes = s.as_bytes();
        let t_bytes = t.as_bytes();
        let mut maxLen = 0;
        let mut total = 0;
        let mut left  = 0;
        for right in 0..s.len(){
            total += (s_bytes[right] as i32 - t_bytes[right] as i32).abs();
            while total > max_cost{
                total-= (s_bytes[left] as i32 - t_bytes[left] as i32).abs();
                left += 1;
            }
            maxLen = maxLen.max((right - left + 1) as i32);
        }
        maxLen
    }
}